subprojects/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompile.java

```diff
@@ -93,7 +93,7 @@ protected void compile() {
 
     private DefaultGroovyJavaJointCompileSpec createSpec() {
         DefaultGroovyJavaJointCompileSpec spec = new DefaultGroovyJavaJointCompileSpecFactory(compileOptions).create();
-        spec.setSourceFiles(getSource().getFiles());
+        spec.setSourceFiles(getSource());
```

Comment:
Won't this result in the `FileCollection` being resolved multiple times? Like when we check if it's empty?

Reply:
Not after the `NormalizingCompiler` replaced it with an `ImmutableSet`. And we cannot resolve it earlier, for the bug reported in https://github.com/gradle/gradle/issues/5448.